plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -299,7 +297,7 @@ public ConnectorPageSource createPageSource(
         List<IcebergColumnHandle> readColumns = dataPageSource.getReaderColumns()
                 .map(readerColumns -> readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toList()))
                 .orElse(requiredColumns);
-        DeleteFilter<TrinoRow> deleteFilter = new TrinoDeleteFilter(
+        TrinoDeleteFilter deleteFilter = new TrinoDeleteFilter(
```

Comment:
I think this is a good short term improvement but I think we should try to avoid keeping all the deleted row numbers in memory if possible.

If we can assume splits for the same file will always come in together, and in order we could use a streaming/iterative read approach.
- First Split for a file arrives, initialize a DeleteFilter
- Second Split arrives for the same file, reuse the existing DeleteFilter
- First Split arrives for a different file, close the old DeleteFilter and open a new one for the new file
- repeat

But I'd want some validation from @findepi on if we can rely on ordering like that

Reply:
A single delete filter file can affect any data file in the partition. So I think we need to load all filters for the partition.
(Unless we open them ahead of time, Analyse them and the determine which ones affect which files)

Reply:
They should be filtered using the min/max values of the path column, so only some of the partition's filters are fully opened

Reply:
This is for positional deletes specifically

Reply:
Thanks @alexjo2144 . Are you saying that is already happening or something we need to add?